v23/context/context_test.go

```diff
@@ -320,6 +322,27 @@ func TestRootCancelChain(t *testing.T) {
 	}
 }
 
+func TestRootCancelGoroutineLeak(t *testing.T) {
+	rootCtx, rootcancel := context.RootContext()
+	const iterations = 1000
+	for i := 0; i != iterations; i++ {
+		_, cancel := context.WithRootCancel(rootCtx)
+		cancel()
+	}
+	buf := make([]byte, 2<<20)
+	buf = buf[:runtime.Stack(buf, true)]
+	count := 0
+	for _, g := range strings.Split(string(buf), "\n\n") {
```

Comment:
this is likely going to be flaky, at least in terms of getting to 0. You may want to just check that some, say a third, of the goroutines have exited since there's no way any of them would exit without your change for this test.